packages/datadog-plugin-restify/test/index.spec.js

```diff
@@ -85,7 +85,16 @@ describe('Plugin', () => {
         it('should run handlers in the request scope', done => {
           const server = restify.createServer()
           const interval = setInterval(() => {
-            next && next() && clearInterval(interval)
+            if (next) {
+              clearInterval(interval)
+              const _next = next
+              next = null
+              try {
+                _next()
```

Comment:
the main thing: next() can now throw, so wrap in try-catch.

we also clear out next when firing it, which is something we didn't do, leading to a lot of not-related `Error [ERR_HTTP_HEADERS_SENT]: Cannot render headers after they are sent to the client` messages when tests broke due. this makes debugging test issues significantly easier & guards against possible race conditions.

Reply:
Why is it needed to change this test? It's not supposed to error.

Reply:
I've simplified this somewhat & removed the try catch.

Some change was still required. We used to not actually call `clearInterval()` as `next()` did not return truthy. This had no impact as the restify suite was already finishing/shutting down. But now that there is another test in the suite, which runs latter, having the interval firing & trying to re-send a reply caused test failures.